### ISSUE TITLE
Intellisense for components are not available

### ISSUE THREAD

**Issue by kongku** [nuxt/content#2586]
### Environment

- Operating System: Windows_NT
- Node Version:     v20.10.0
- Nuxt Version:     3.10.3
- CLI Version:      3.10.1
- Nitro Version:    2.9.3
- Package Manager:  pnpm@8.15.1
- Builder:          -
- User Config:      modules
- Runtime Modules:  @nuxt/content@2.12.1
- Build Modules:    -

### Reproduction

no

### Describe the bug

With the exception of the ContentRendererMarkdown component, all components are written using the options style and defineComponent function without lang="ts", and the relevant declares in the components.d.ts file in the .nuxt directory are as follows, which loads not a d.ts file, but a .vue file, causing vscode's intellisense to fail on these components。
![image](https://github.com/nuxt/content/assets/29189288/104e4333-97a4-4c42-80cb-7869d77c100c)
![image](https://github.com/nuxt/content/assets/29189288/9f618f1c-45be-4428-b25d-58bc368cfe8c)
![image](https://github.com/nuxt/content/assets/29189288/d0ac7c6a-472f-43da-9fa0-a1d1c641a73d)

Looking at the source code, there is a lang="ts". I tried to compile the source code, same thing. Is this a bug in @nuxt/module-builder?
![image](https://github.com/nuxt/content/assets/29189288/5c5d7643-2b7d-46db-916e-bcc7deac0038)

So I need to add the declarations for these components myself。
```.d.ts
declare module 'vue' {
  export interface GlobalComponents {
    ContentDoc: typeof import('./node_modules/@nuxt/content/dist/runtime/components/ContentDoc')['default']
    ContentList: typeof import('./node_modules/@nuxt/content/dist/runtime/components/ContentList')['default']
    ContentNavigation: typeof import('./node_modules/@nuxt/content/dist/runtime/components/ContentNavigation')['default']
    ContentQuery: typeof import('./node_modules/@nuxt/content/dist/runtime/components/ContentQuery')['default']
    ContentRenderer: typeof import('./node_modules/@nuxt/content/dist/runtime/components/ContentRenderer')['default']
    ContentSlot: typeof import('./node_modules/@nuxt/content/dist/runtime/components/ContentSlot')['default']
    DocumentDrivenEmpty: typeof import('./node_modules/@nuxt/content/dist/runtime/components/DocumentDrivenEmpty')['default']
    DocumentDrivenNotFound: typeof import('./node_modules/@nuxt/content/dist/runtime/components/DocumentDrivenNotFound')['default']
    Markdown: typeof import('./node_modules/@nuxt/content/dist/runtime/components/Markdown')['default']
  }
}
export {}
```

**Comment by Coiggahou2002** [comment]
Normally things may happens like this (let's say your project directory is 'src'):
1. The TypeScript language server in your VS Code loads the `src/tsconfig.json`
2. `src/tsconfig.json` extends `src/.nuxt/tsconfig.json`
3. `src/.nuxt/tsconfig.json` includes `src/.nuxt/nuxt.d.ts`
4. `src/.nuxt/nuxt.d.ts` references the `src/.nuxt/components.d.ts`
5.  `src/.nuxt/components.d.ts` who declares an `interface _GlobalComponents {...}` which holds the auto-imported components including `<ContentDoc/>`
6. There should be something like ` 'ContentDoc': typeof import("../node_modules/@nuxt/content/dist/runtime/components/ContentDoc.vue")['default']` inside `components.d.ts`
7. Also, there should be a `ContentDoc.vue.d.ts` who goes along with the `ContentDoc.vue` in `node_modules/@nuxt/content/dist/runtime/components/` which will be imported when the `.vue` file being imported


```mermaid
stateDiagram-v2
  A: src/tsconfig.json
  B: src/.nuxt/tsconfig.json
  C: src/.nuxt/nuxt.d.ts
  D: src/.nuxt/components.d.ts
  A -->B
  B --> C
  C --> D
```


In conclusion, I don't think this is a bug related to Nuxt Content or its build system. It's more likely related to the Vue Plugin you used in your VS Code or the TypeScript language server in your VS Code. If every steps above looks good, please have a try to [restart your TS server](https://stackoverflow.com/questions/64454845/where-is-vscodes-restart-ts-server).
